### PR TITLE
PLT-424: Add a budget test for PlutusTx.Show

### PR DESCRIPTION
**Pull request by zliu41**
Also optimized the `Show Integer` instance as @michaelpj suggested (if negative, take the inverse first), which is about 20% more efficient because it no longer checks the sign of each digit.

### CODE REVIEW

**Review by michaelpj — APPROVED** [review]
I was also thinking it would be nice to just have two tests for e.g. a simple function in a tail-recursive and non-tail recursive version.

**Comment by zliu41** [comment]
> a simple function in a tail-recursive and non-tail recursive version.

Yeah, good idea.

**Review by effectfully — APPROVED** [review]

**Comment by zliu41** [comment]
Sneaking in a commit (1f292c7) that changes some long names to symbols.

File: plutus-tx/src/PlutusTx/List.hs [hunk 1]
```diff
@@ -172,7 +172,7 @@ splitAt n xs
     go :: Integer -> [a] -> ([a], [a])
     go _ []     = ([], [])
     go m (y:ys)
-      | m `Builtins.equalsInteger` 1 = ([y], ys)
+      | m == 1 = ([y], ys)
       | otherwise = case go (Builtins.subtractInteger m 1) ys of
```
Comment (bezirg) [inline plutus-tx/src/PlutusTx/List.hs:176]: i think this could also be made `(m - 1)`

Comment (zliu41) [reply plutus-tx/src/PlutusTx/List.hs:176]: This module can't import `PlutusTx.Numeric` because it would cause circular dependency.

**Pull request merged**